graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/ScrollResultES6.java

```diff
@@ -0,0 +1,164 @@
+
+    private final JestClient jestClient;
+    private final ObjectMapper objectMapper;
+    private SearchResult initialResult;
+    private final String scroll;
+    private final List<String> fields;
+    private final String queryHash; // used in log output only
+    private final long totalHits;
+
+    private String scrollId;
+    private int chunkId = 0;
+
+    public interface Factory {
+        ScrollResultES6 create(SearchResult initialResult, @Assisted("query") String query, @Assisted("scroll") String scroll, List<String> fields);
+    }
+
+
+    @AssistedInject
+    public ScrollResultES6(JestClient jestClient, ObjectMapper objectMapper, @Assisted SearchResult initialResult, @Assisted("query") String query, @Assisted List<String> fields) {
+        this(jestClient, objectMapper, initialResult, query, DEFAULT_SCROLL, fields);
+    }
+
+    @AssistedInject
+    public ScrollResultES6(JestClient jestClient, ObjectMapper objectMapper, @Assisted SearchResult initialResult, @Assisted("query") String query, @Assisted("scroll") String scroll, @Assisted List<String> fields) {
+        super(query, null, initialResult.getJsonObject().path("took").asLong());
+        this.jestClient = jestClient;
+        this.objectMapper = objectMapper;
+        this.initialResult = initialResult;
+        this.scroll = scroll;
+        this.fields = fields;
+        totalHits = initialResult.getTotal();
+        scrollId = getScrollIdFromResult(initialResult);
+
+        final Md5Hash md5Hash = new Md5Hash(getOriginalQuery());
+        queryHash = md5Hash.toHex();
+
+        LOG.debug("[{}] Starting scroll request for query {}", queryHash, getOriginalQuery());
+    }
+
+    @Override public ScrollChunk nextChunk() throws IOException {
+
+        final JestResult search;
+        final List<ResultMessage> hits;
+        if (initialResult == null) {
+            search = getNextScrollResult();
+            hits = StreamSupport.stream(search.getJsonObject().path("hits").path("hits").spliterator(), false)
+                    .map(hit -> ResultMessage.parseFromSource(hit.path("_id").asText(),
+                            hit.path("_index").asText(),
+                            objectMapper.convertValue(hit.get("_source"), TypeReferences.MAP_STRING_OBJECT)))
+                    .collect(Collectors.toList());
+        } else {
+            // make sure to return the initial hits, see https://github.com/Graylog2/graylog2-server/issues/2126
+            search = initialResult;
+            hits = initialResult.getHits(Map.class, false).stream()
+                .map(hit -> ResultMessage.parseFromSource(hit.id, hit.index, (Map<String, Object>)hit.source))
+                .collect(Collectors.toList());
+            this.initialResult = null;
+        }
+
+        if (hits.size() == 0) {
+            // scroll exhausted
+            LOG.debug("[{}] Reached end of scroll results.", queryHash, getOriginalQuery());
```

Comment:
I know this bug was in here before, but there's a placeholder missing in the format string. I think we should fix that.